[YDS-CheckBox] companion에 있는 속성의 이름변경, Disabled 여부에 따라 폰트변경 추가, 바인딩 어댑터로 이전

Gael-Android opened the pull request:
컴패니언의 상수들을 모두 노션에 나와있는 기본 속성으로 변경하였습니다.
binding을 통해서 size에 접근할 때 이제 CheckBox.small, CheckBox.medium, CheckBox.large로 접근할 수 있습니다.

Gael-Android committed:
체크박스 companion에 있는 속성의 이름변경

컴패니언의 상수들을 모두 노션에 나와있는 기본 속성으로 변경하였습니다.
binding을 통해서 size에 접근할 때 이제 CheckBox.small, medium, large로 접근할 수 있습니다.

islandparadise14 commented:
컨플릭 났음

islandparadise14 commented:
이름을 통일하는건 좋은데 관념적으로 const는 대문자로 해야하지 않을까요?

Gael-Android committed:
상수명 대문자, isDisabled 여부에 따른 small size button의 폰트변화

상수명이 small=>SMALL로 변경됨.
size가 small인 버튼이 isDisabled면 button4고 아니면 button3 인것을 수정함.

Gael-Android committed:
Update AndroidManifest.xml

Gael-Android committed:
Merge branch 'develop' into feature/gael/checkbox

Gael-Android committed:
Merge branch 'develop' into feature/gael/checkbox

Gael-Android committed:
Merge branch 'feature/gael/checkbox' of https://github.com/yourssu/YDS-Android into feature/gael/checkbox

Gael-Android committed:
액티비티 내부의 상수들 변경

Gael-Android committed:
setState에 size를 변경할 수 있게 함.

매니페스트 수정함.

islandparadise14 commented on app/yds-ui-tester/src/main/AndroidManifest.xml:
```diff
@@ -9,15 +9,17 @@
-        <activity android:name=".ToggleActivity"/>
-        <activity android:name=".DividerActivity" />
-        <activity android:name=".DesignSystemListActivity">
+        <activity android:name=".CheckBoxActivity">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+        </activity>
```
이런건 테스트 할때 변경했더라도 올릴떄는 원복해주세요

islandparadise14 commented:
<img width="576" alt="스크린샷 2021-09-15 오전 2 16 15" src="https://user-images.githubusercontent.com/39309953/133303646-d4748efb-6983-4084-9928-6d6a66872c9d.png">
지금은 본인 브렌치에만 커밋이 찍히지만 develop에 머지하는 순간 그냥 커밋들이 나열됩니다
그때 `setState에 size를 변경할 수 있게 함.` 이것만 보고 체크박스를 연상하긴 힘들겠죠?
조금 귀찮더라도 어디에 무엇을 변경함 같이 쓰면 좋겠습니다
`checkbox의 setState를 size 변경할수 있도록 수정` 이런식으로요

islandparadise14 commented:
잘했는데 manifest만 원복하면 바로 approve 드림

Gael-Android committed:
매니페스트 원상복구, 레이아웃 변경

Gael-Android committed:
체크박스 처음 클릭위치에서 뷰 크기보다 벗어나면 터치 안되게 만듬

체크박스의 onTouchEvent를 수정함.

islandparadise14 commented:
컨플릭 ㅠ

Gael-Android committed:
Merge branch 'develop' into feature/gael/checkbox

Gael-Android committed:
체크박스 바인딩 어댑터로 이전, attrs 파일 삭제, 테스트 액티비티 파일 수정. 레이아웃 파일의 아이콘의 바인딩 처리 제거.

문제발생(1): 처음 실행될 때 레이아웃에서 정해둔 기본값이 정해짐. 클릭을 하게 되면 사용자가 XML에 지정한 값이 정상적으로 돌아옴.

문제해결(1): 그냥 체크박스 레이아웃에 있는 iconView에 데이터 바인딩으로 처리하는 부분을 삭제함.

Gael-Android committed:
setSelected 바인딩 어댑터 오타 수정, 세터 파라미터 이름 변경

Gael-Android committed:
Merge branch 'develop' into feature/gael/checkbox

Gael-Android committed:
Merge branch 'develop' into feature/gael/checkbox

Gael-Android committed:
인포트가 안되어 있어서 제대로 다시 연결함

islandparadise14 commented on DesignSystem/src/main/java/com/yourssu/design/system/atom/CheckBox.kt:
```diff
@@ -25,53 +27,46 @@ class CheckBox @JvmOverloads constructor(
         setDrawable()
+        setSizeState()
     }
 
     private fun setTotalColor() {
-        when (!isDisabled) {
+        when (isDisabled) {
             true ->
+                changeTotalColor(R.color.buttonDisabled)
+            false ->
                 if (isSelected)
                     changeTotalColor(R.color.buttonPoint)
                 else
                     changeTotalColor(R.color.buttonNormal)
-            false ->
```
이런건 true false 두개밖에 없으니 굳이 when으로 할 필욘 없어 보이는데 ㅋㅋ 일단 상관없
